substrate/frame/revive/src/gas.rs

```diff
@@ -168,25 +168,18 @@ impl<T: Config> GasMeter<T> {
 		}
 	}
 
-	/// Create a new gas meter by removing gas from the current meter.
-	///
-	/// # Note
+	pub(crate) fn take_all(&mut self) -> Self {
+		let gas_left = self.gas_left;
+		self.gas_left -= gas_left;
+		GasMeter::new(gas_left)
+	}
```

Comment:
- No need for the `crate` as this is not a public module. Lates keep it consistent.
- I think the name should reflect we are creating a nested meter.
- Needs some doc comment mentioning why we would use this function over just `nested`.

```suggestion
	pub fn nested_take_all(&mut self) -> Self {
		let gas_left = self.gas_left;
		self.gas_left -= gas_left;
		GasMeter::new(gas_left)
	}
```